api_app/api/routes/airlock.py

```diff
@@ -90,19 +203,35 @@ async def create_airlock_review(airlock_review_input: AirlockReviewInCreate, air
     elif airlock_review.reviewDecision.value == AirlockReviewDecision.Rejected:
         review_status = AirlockRequestStatus.RejectionInProgress
 
-    updated_airlock_request = await update_and_publish_event_airlock_request(airlock_request=airlock_request, airlock_request_repo=airlock_request_repo, user=user, new_status=review_status, workspace=workspace, airlock_review=airlock_review)
+    # If there was a VM created for the request, clean it up as it will no longer be needed
+    # In this request, we aren't returning the operations for clean up of VMs,
+    # however the operations still will be saved in the DB and displayed on the UI as normal.
+    _ = await delete_review_user_resources(
```

Comment:
Should this maybe be moved to after the update airlock request completes, as if it fails for whatever reason, the review VM will have been deleted but the request would still be in a review status

Reply:
A very good point! Will do that